### PR TITLE
[SR-66] Add Top Products by line item totals reports

### PR DESCRIPTION
**Pull request by mbajur**

### CODE REVIEW

File: backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          attr_reader :currency
+
+          def top
+            return nil if params[:top].nil?
+
+            params[:top].to_i
+          end
+
+          def by_top(variants)
+            return variants unless top
```
Comment (przemosk) [inline backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb]: @mbajur when method `top` will return `true` or `false` ?

File: backend/app/serializers/spree/admin/reports/top_products_by_line_item_totals/csv_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+module Spree
+  module Admin
+    module Reports
+      module TopProductsByUnitSold
+        class CsvSerializer
+          def call(objects)
+            CSV.generate do |csv|
+              csv << %w[sku line_item_totals]
+
+              objects.each do |sku, line_item_totals|
```
Comment (przemosk) [inline backend/app/serializers/spree/admin/reports/top_products_by_line_item_totals/csv_serializer.rb]: @mbajur wdyt about change `each` to `find_each` ?

File: backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb (hunk 1 above)
Comment (mbajur) [reply backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb]: @piotrleniec-spark i think that was your change when you were working on POC for SQL-based solution. Do you remember the reason?

File: backend/app/serializers/spree/admin/reports/top_products_by_line_item_totals/csv_serializer.rb (hunk 2 above)
Comment (mbajur) [reply backend/app/serializers/spree/admin/reports/top_products_by_line_item_totals/csv_serializer.rb]: `objects` is an array, not ActiveRecord collection so we can't use it in here.

File: backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb (hunk 1 above)
Comment (mbajur) [reply backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb]: image

never, it's just a shorthand notation for `top.present?`

**Pull request merged**